### PR TITLE
reference: add info for error codes

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add info for error codes.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/2889
- Other reference link(s):

### CODE REVIEW

File: reference/error-codes.md [hunk 1]
```diff
@@ -43,83 +45,94 @@ In addition, TiDB has the following unique error codes:
...
+    The table schema version is outdated. TiDB uses the F1 online schema change algorithm to execute DDL. When the table schema version of TiDB server is older than that of the entire system, this error is returned if you execute a SQL statement.
+
+    When this error occurs, check the network between the TiDB server and PD Leader.
 
 * Error Number: 8028
 
     The table schema has changed.
+    TiDB does not support table lock, which is called metadata lock in MySQL and might be called intention lock in other databases.
+
+    When a transaction is executed, the transaction cannot perceive the table schema changes. Therefore, when committing a transaction, TiDB checks the table schema related the transaction. If the related table schema has changed during the execution, the transaction commit will fail.
```
Comment (TomShawn) [inline reference/error-codes.md:105]:
```suggestion
    When a transaction is executed, the transaction cannot recognize the table schema changes. Therefore, when committing a transaction, TiDB checks the table schema related the transaction. If the related table schema has changed during the execution, the transaction commit will fail and this error is returned.
```

File: reference/error-codes.md [hunk 2]
```diff
@@ -43,83 +45,94 @@ In addition, TiDB has the following unique error codes:
...
-
-* Error Number: 8036
-
-    Invalid `time` format.
+    Invalid `year` value is used. The valid range of `year` is (1901, 2155).
 
 * Error Number: 8037
 
-    Invalid `week` format.
+    Invalid `mode` format is used in the `week` function. `mode` must be 1 digit in [0, 7].
```
Comment (TomShawn) [inline reference/error-codes.md:131]:
```suggestion
    Invalid `mode` format is used in the `week` function. `mode` must be 1 digit within [0, 7].
```

**Comment by ran-huang** [comment]
Comments addressed. PTAL again~ @TomShawn

**Pull request merged**